build-tools/packages/build-cli/src/library/dangerfile.ts

```diff
@@ -47,6 +65,26 @@ export async function dangerfile(): Promise<void> {
 	// there were actual changes to the bundle sizes.  In other cases, we don't post a
 	// message and danger will delete its previous message
 	if (result.comparison === undefined || !bundlesContainNoChanges(result.comparison)) {
+		// Check for bundle size regression
+		const sizeCheck =
+			result.comparison?.some((bundle: BundleComparison) => {
+				const totalMetric = bundle.commonBundleMetrics[totalSizeMetricName];
```

Comment:
Oh I see what Tony is talking about now, didn't realize this was doubly nested.  So something like this would probably be a little better.

```ts
Object.entries(bundle.commonBundleMetrics).filter(([key, val]) => key !== totalSizeMetricName).some(...)
```

Reply:
I wasn't talking about your change. i was talking about a limitation of the tooling, and an existing bug where we add the sizes of all bundles together which is invalid. The sizes cannot be added, they are only valid independently, as each bundle will have its own copies of shared things, like telemetry.

Reply:
![image](https://github.com/microsoft/FluidFramework/assets/1951092/6fd5b31a-e044-41b1-a40c-d05b66cc778d)

Reply:
the things circled in red are invalid, as they add the bundles together, which shouldn't be done

Reply:
I switched to looking through the bundle metrics independently and not including the total. I looked at how [the table is generated](https://github.com/microsoft/FluidFramework/blob/92326618f47840eac4e8df796c93458f29855815/build-tools/packages/bundle-size-tools/src/ADO/getCommentForBundleDiff.ts#L57C1-L67C1) as reference.